zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -155,44 +153,31 @@ public OrchestratorZoneResponse deleteZone(String zoneName) {
         return response;
     }
 
-    private ConnectionDetails buildConnectionDetails(String zoneName, OrchestratorZoneEntity orchestratorZone,
-                                                     String zoneUri) {
+    private ConnectionDetails buildConnectionDetails(OrchestratorZoneEntity orchestratorZone) {
         ConnectionDetails connectionDetails = new ConnectionDetails();
-        connectionDetails.setUri(zoneUri);
-        connectionDetails.setIssuerId(zoneUri + "/oauth/token");
+        connectionDetails.setUri(constructUri(orchestratorZone.getSubdomain(), uaaUrl, ""));
+        connectionDetails.setIssuerId(constructUri(orchestratorZone.getSubdomain(), issuerUri, "oauth/token"));
         connectionDetails.setSubdomain(orchestratorZone.getSubdomain());
         connectionDetails.setDashboardUri(uaaDashboardUri);
         OrchestratorZoneHeader zoneHeader = new OrchestratorZoneHeader(X_IDENTITY_ZONE_ID, orchestratorZone.getIdentityZoneId());
         connectionDetails.setZone(zoneHeader);
         return connectionDetails;
     }
 
-    private String getZoneUri(String subdomain, String uaaUri) {
-        URI uaaUriObject = URI.create(uaaUri);
-        String currentUAAHostName = uaaUriObject.getHost();
-        String replacement = subdomain + "." + currentUAAHostName;
-        if (!hasLength(subdomain)) {
-            replacement = currentUAAHostName;
-        };
-        URI newUAARoute = URI
-            .create(uaaUriObject.toString().replace(currentUAAHostName, replacement));
-        return newUAARoute.toString();
+    private String constructUri(String subDomain, String baseUrl, String path) {
+        URI uri = URI.create(baseUrl);
+        String hostToUse = uri.getHost();
+        if (hasText(subDomain)) {
+            hostToUse = subDomain + "." + hostToUse;
+        }
+        return UriComponentsBuilder.fromUri(uri).host(hostToUse).pathSegment(path).build().toUriString();
```

Comment:
does pathSegment() method appends "/" after base url as well? because in line 159, it is invoked like this, 
constructUri(orchestratorZone.getSubdomain(), issuerUri, "oauth/token").

Reply:
Yes, It will append "/" after base url.